UM-USElab/gradecraft-development
# 2941 resubmission after grading

### Status
**READY**

### Description
This PR fixes a bug where submissions that had grades that were marked as `in_progress` were still identified as `ungraded` on the grading status page. It also moves the `open_for_editing?(assignment)` method over to a proctor. It also blocks students from submitting a resubmission unless their grade has actually been returned to them by the instructor. 


### Todos


### Migrations
NO

### Impacted Areas in Application
List general components of the application that this PR will affect:

* Grading Status
* Resubmissions

### To Test
_In Progress Grades_
* Add a grade for an ungraded Submission, mark it as "In Progress"
* In master branch, you should still see it appear in the Ungraded Section (but also under In Progress)
• In this branch, it should only appear under In Progress

_Resubmitting Assignment Conditions_
• Confirm that on the student side, they cannot update/resubmit the assignment before they've actually gotten a grade back. 

======================
Closes #3131
